Added an expiration property

iherman opened the pull request:
Releated to the issues and comments in https://github.com/w3c/controller-document/pull/92#discussion_r1749537440: the `expires` property was present in the DI version of the verification methods, and remained commented out in the controller spec. This is probably a property to be re-instated.

Note that the current text (which I took by removing some comment lines) is identical to `revoked`. What differentiates these two? Per the spec, nothing...

A separate PR will be raised on the DI Vocabulary to make this valid.


<!--
    This comment and the below content is programmatically generated.
    You may add a comma-separated list of anchors you'd like a
    direct link to below (e.g. #idl-serializers, #idl-sequence):

    Don't remove this comment or modify anything below this line.
    If you don't want a preview generated for this pull request,
    just replace the whole of this comment's content by "no preview"
    and remove what's below.
-->
***
<a href="https://pr-preview.s3.amazonaws.com/w3c/controller-document/pull/96.html" title="Last updated on Sep 12, 2024, 6:41 AM UTC (7041f2c)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/w3c/controller-document/96/1709c9e...7041f2c.html" title="Last updated on Sep 12, 2024, 6:41 AM UTC (7041f2c)">Diff</a>

longpd reviewed (approved):
Good catch Ivan. Approve this change.

iherman commented:
The issue was discussed in a [meeting](https://www.w3.org/2017/vc/WG/Meetings/Minutes/2024-09-11-vcwg#section4-1) on 2024-09-11

- no resolutions were taken

<details><summary><i>View the transcript</i></summary>

#### 4.1. Added an expiration property (pr controller-document#96)

_See github pull request [controller-document#96](https://github.com/w3c/controller-document/pull/96)._




**Brent Zundel:** ivan, you have raised this PR, has been open for 4 hours, talk us through the changes.  

**Ivan Herman:** dlongley updated the context file which I put in another PR, by looking at that I realized that with all the changes, the expire property is not mentioned as a potential property for verification methods, I believe it was in DI origin but got commented out in controller document, I believe that is a transition bug, I removed the comments to put it back.  
… two comments, if this gets accepted I will have to make some changes in the vocabulary, and also the text that is there, which I did not write, is the same as for another property.  
… I understand the difference between the two but the specification should make it clearer.  

**Manu Sporny:** +1 to the PR, the differences are: expires is set before the thing expires as just a window, the revoked thing is that you must pay attention to revoked as the key was deactivated on a certain date for a very specific reason, while you should pay attention to expires.  

**Ivan Herman:** I will modify the PR to make that difference clear and we can come back to it.  

**Michael Jones:** the history of that comment and similar comments is that when I was sorting out what was common between controller document section in vc-jose-cose and that section in DI, anything that was not in both I put in comments, only put in stuff from both documents.  

</details>

iherman commented:
Per the discussion on the meeting (see https://www.w3.org/2017/vc/WG/Meetings/Minutes/2024-09-11-vcwg#section4-1) I have modified the text, saying that after the 'revoked' date the key _MUST NOT_ be used.

@msporny did I understand well what you said yesterday? Reviewers, please consider this in your final opinion.

iherman commented:
@msporny any reason why not merge this, too?

msporny commented:
Normative, multiple reviews, no changes requested, no objections, merging.
